src/main/scala/com/fulcrumgenomics/util/DelimitedDataParser.scala

```diff
@@ -0,0 +1,128 @@
+  * Represents a row of parsed data.  Provides methods for accessing values in a type-safe
+  * way either via apply() methods for non-optional fields or via get for optional fields.
+  */
+class Row (private val headerIndices: Map[String,Int], private val fields: Array[String], val trim: Boolean) {
+  /* Internal method to pull a value out of the field array by index and trim it if requested. */
+  private def value(index: Int) = {
+    if (index > fields.length-1) throw new IndexOutOfBoundsException(s"Invalid column index supplied: ${index}")
+    val string = fields(index)
+    if (trim) string.trim else string
+  }
+
+  /** Fetches a value of the desired type by 0-based column index. */
+  def apply[A](index: Int)(implicit tag: ru.TypeTag[A]): A = {
+    val c = ReflectionUtil.typeTagToClass(tag)
+    ReflectionUtil.constructFromString(c, c, value(index)).get.asInstanceOf[A]
+  }
+
+  /** Fetches a value of the desired type by column name. */
+  def apply[A](column: String)(implicit tag: ru.TypeTag[A]): A = apply(headerIndices.apply(column))
+
+  /** Gets a value from the column with the specified index. If the value is null or empty returns None. */
+  def get[A](index: Int)(implicit tag: ru.TypeTag[A]): Option[A] = {
+    val string = value(index)
+    if (string == null || string.isEmpty) {
+      None
+    }
+    else {
+      val c = ReflectionUtil.typeTagToClass(tag)
+      Some(ReflectionUtil.constructFromString(c, c, value(index)).get.asInstanceOf[A])
+    }
+  }
+
+  /** Gets a value from the specified column. If the value is null or empty returns None. */
+  def get[A](column: String)(implicit tag: ru.TypeTag[A]): Option[A] = get(headerIndices.apply(column))
+}
+
+
+object DelimitedDataParser {
+  val DefaultDelimiter: Char = '\t'
+  val DefaultBlankPolicy: Boolean = true
+  val DefaultTrim : Boolean = true
+
+  /** Constructs a DelimitedDataParser for a path. */
+  def apply(path: FilePath, delimiter: Char): DelimitedDataParser =
+    new DelimitedDataParser(Io.toSource(path).getLines(), delimiter=delimiter)
+
+  /** Constructs a DelimitedDataParser for a path. */
+  def apply(lines: Traversable[String], delimiter: Char): DelimitedDataParser =
+    new DelimitedDataParser(lines, delimiter=delimiter)
+}
+
+/**
+  * A parser for files of text columns delimited by some character (e.g. tab-delimited or csv).
+  *
+  * @param lines the lines from the file
+  * @param delimiter the delimiter between columns
+  * @param ignoreBlankLines whether blank lines should be ignored
+  * @param trimFields whether individual fields should have their String values trimmed
+  */
+class DelimitedDataParser(lines: TraversableOnce[String],
+                          val delimiter: Char = DelimitedDataParser.DefaultDelimiter,
+                          val ignoreBlankLines: Boolean = DelimitedDataParser.DefaultBlankPolicy,
+                          val trimFields: Boolean = DelimitedDataParser.DefaultTrim) extends Iterator[Row] with LazyLogging {
+
```

Comment:
It may be useful if the delimiter were a regex, for example, if we we want to ignore commas within quotes we could use `"\t(?=([^\"]*\"[^\"]*\")*[^\"]*$)".r`.